lib/grr.js

```diff
@@ -1399,13 +1403,244 @@ function stepListCRs(arg, cb) {
+                        console.log('patch:');
+                        console.log(buf.toString('utf8'));
+                        if (arg.dryRun) {
+                            console.log('Dry run, will not prompt for review.');
+                            return next();
+                        }
+                        var review_desc = extsprintf(
+                            'Do you want to give a \'%s\' +1 for this review?',
+                            arg.reviewArg);
+                        // prompt for user input, and apply CR
+                        var prompt_schema = {
+                            properties: {
+                                answer: {
+                                    description: review_desc,
+                                    pattern: /[y|n]/,
+                                    message: 'y or n will do!',
+                                    required: true
+                                }
+                            }
+                        };
+                        prompt.colors = false;
+                        prompt.message = '';
+                        prompt.start();
+                        prompt.get(
+                            prompt_schema,
+                            function user_input(prompt_err, result) {
+                                if (prompt_err) {
+                                    next(new VError(
+                                        prompt_err,
+                                        'problem trying to prompt user'));
+                                    return;
+                                }
+                                if (result.answer == 'y') {
+                                    console.log(extsprintf(
+                                        'Granting a \'%s\' +1 to this review',
+                                        arg.reviewArg));
+                                    common.forkExecWaitAndLog({
+                                        argv: ['ssh', '-q',
+                                            '-o', 'StrictHostKeyChecking=no',
+                                            '-o', 'UserKnownHostsFile=/dev/null',
+                                            '-p', '29418',
+                                            arg.gerritUsername + '@cr.joyent.us',
+                                            'gerrit', 'review',
+                                            '--project', cr_res.project,
+                                            '--label', arg.reviewArg + '=+1',
+                                            cr_res.currentPatchSet.revision
+                                        ],
+                                        log: arg.log
+                                    // eslint-disable-next-line no-unused-vars
+                                    }, function (fork_err, review_res) {
+                                        if (fork_err) {
+                                            next(fork_err);
```

Comment:
If there is a `fork_err`, then this `next(fork_err)` will be called *and* the `next()` a few lines down will already have been called.

Reply:
Thanks, I've sorted that out by moving the later call to `next()` into an `else` clause on that if-statement.